geocity/tests/submissions/functional/test_submission_classify.py

```diff
@@ -350,27 +351,118 @@ def _get_approval(self):
+        self,
+    ):
+        form = factories.FormFactory(
+            validation_document=True,
+            validation_document_required_for=forms_models.Form.VALIDATION_DOCUMENT_REQUIRED_FOR_REFUSAL,
+        )
         self.validation.submission.forms.set([form])
         response = self._get_approval()
         self.assertNotContains(response, "validation_pdf")
 
-    def test_classify_submission_with_any_object_requiring_validation_doc_shows_file_field(
+    def test_classify_reject_submission_with_required_validation_doc_for_refusal_shows_file_field(
+        self,
+    ):
+        form = factories.FormFactory(
+            validation_document=True,
+            validation_document_required_for=forms_models.Form.VALIDATION_DOCUMENT_REQUIRED_FOR_REFUSAL,
+        )
+        self.validation.submission.forms.set([form])
+        response = self._get_refusal()
+        self.assertContains(response, "validation_pdf")
+
+    def test_classify_approve_submission_with_required_validation_doc_for_approval_and_refusal_shows_file_field(
         self,
     ):
-        form1 = factories.FormFactory(requires_validation_document=True)
-        form2 = factories.FormFactory(requires_validation_document=False)
+        form = factories.FormFactory(
+            validation_document=True,
+            validation_document_required_for=forms_models.Form.VALIDATION_DOCUMENT_REQUIRED_FOR_APPROVAL_AND_REFUSAL,
+        )
+        self.validation.submission.forms.set([form])
+        response = self._get_approval()
+        self.assertContains(response, "validation_pdf")
+
+    def test_classify_reject_submission_with_required_validation_doc_for_approval_and_refusal_shows_file_field(
+        self,
+    ):
+        form = factories.FormFactory(
+            validation_document=True,
+            validation_document_required_for=forms_models.Form.VALIDATION_DOCUMENT_REQUIRED_FOR_APPROVAL_AND_REFUSAL,
+        )
+        self.validation.submission.forms.set([form])
+        response = self._get_refusal()
+        self.assertContains(response, "validation_pdf")
+
+    def test_classify_approve_submission_with_any_object_requiring_validation_doc_shows_file_field(
+        self,
+    ):
+        form1 = factories.FormFactory(validation_document=True)
+        form2 = factories.FormFactory(validation_document=False)
         self.validation.submission.forms.set([form1, form2])
         response = self._get_approval()
         self.assertContains(response, "validation_pdf")
+
+    def test_classify_approve_submission_with_default_validation_text_shows_text(
+        self,
+    ):
+        default_validation_text = "Mon texte de validation par défaut"
+        form = factories.FormFactory(
+            default_validation_text=default_validation_text,
+        )
+        self.validation.submission.forms.set([form])
+        response = self._get_approval()
+        self.assertContains(response, default_validation_text)
```

Comment:
You might want to add tests that ensures the classify step ("approve/reject") routes works correcty. For instance, reverser the route in the test and check that the submission statuts has changed ?

Reply:
It's already done here https://github.com/yverdon/geocity/blob/d4156d30c5506bc2622d3f3e32d4047833672624/geocity/tests/submissions/functional/test_submission_classify.py#L46-L64 and here https://github.com/yverdon/geocity/blob/d4156d30c5506bc2622d3f3e32d4047833672624/geocity/tests/submissions/functional/test_submission_classify.py#L97-L115

Reply:
Good